server/sonar-server/src/main/java/org/sonar/server/user/UserSession.java

```diff
@@ -92,6 +92,26 @@
   boolean hasProjectPermissionByUuid(String permission, String projectUuid);
 
   /**
+   * Ensures that user implies the specified global or project permission, otherwise throws a {@link org.sonar.server.exceptions.ForbiddenException}.
+   */
+  UserSession checkGlobalOrProjectPermission(String permission, String projectKey);
```

Comment:
These 4 methods must not be added. `UserSession` must have only:
```
boolean hasPermission(String permission);
boolean hasPermissionOnComponent(String permission, String componentUuid);

// and the related methods that raise an exception instead of returning a boolean
UserSession checkPermission(String permission);
UserSession checkPermissionOnComponent(String permission, String componentUuid); 
```